src/AggregateFunctions/SingleValueData.h

```diff
@@ -0,0 +1,394 @@
+
+
+/// For any other value types.
+struct SingleValueDataGeneric final : public SingleValueDataBase
+{
+    static constexpr bool is_compilable = false;
+
+private:
+    using Self = SingleValueDataGeneric;
+    Field value;
+
+public:
+    bool has() const override { return !value.isNull(); }
+    void insertResultInto(IColumn & to) const override;
+    void write(WriteBuffer & buf, const ISerialization & serialization) const override;
+    void read(ReadBuffer & buf, const ISerialization & serialization, Arena *) override;
+
+    bool isEqualTo(const IColumn & column, size_t row_num) const override;
+    bool isEqualTo(const SingleValueDataBase & other) const override;
+    void set(const IColumn & column, size_t row_num, Arena *) override;
+    void set(const SingleValueDataBase & other, Arena *) override;
+
+    bool setIfSmaller(const IColumn & column, size_t row_num, Arena * arena) override;
+    bool setIfSmaller(const SingleValueDataBase & other, Arena *) override;
+    bool setIfGreater(const IColumn & column, size_t row_num, Arena * arena) override;
+    bool setIfGreater(const SingleValueDataBase & other, Arena *) override;
+
+    static bool allocatesMemoryInArena() { return false; }
+};
+
+static_assert(sizeof(SingleValueDataGeneric) <= SingleValueDataBase::MAX_STORAGE_SIZE, "Incorrect size of SingleValueDataGeneric struct");
+
+/// min, max, any, anyLast, anyHeavy, etc...
+template <template <typename, bool...> class AggregateFunctionTemplate, bool unary, bool... isMin>
+static IAggregateFunction *
+createAggregateFunctionSingleValue(const String & name, const DataTypes & argument_types, const Array & parameters, const Settings *)
+{
+    assertNoParameters(name, parameters);
+    if constexpr (unary)
+        assertUnary(name, argument_types);
+    else
+        assertBinary(name, argument_types);
+
+    const DataTypePtr & value_type = unary ? argument_types[0] : argument_types[1];
+    WhichDataType which(value_type);
+#define DISPATCH(TYPE) \
+    if (which.idx == TypeIndex::TYPE) \
+        return new AggregateFunctionTemplate<SingleValueDataFixed<TYPE>, isMin...>(argument_types); /// NOLINT
+    FOR_SINGLE_VALUE_NUMERIC_TYPES(DISPATCH)
+#undef DISPATCH
+
+    if (which.idx == TypeIndex::Date)
+        return new AggregateFunctionTemplate<SingleValueDataFixed<DataTypeDate::FieldType>, isMin...>(argument_types);
+    if (which.idx == TypeIndex::DateTime)
+        return new AggregateFunctionTemplate<SingleValueDataFixed<DataTypeDateTime::FieldType>, isMin...>(argument_types);
+    if (which.idx == TypeIndex::String)
+        return new AggregateFunctionTemplate<SingleValueDataString, isMin...>(argument_types);
+
+    return new AggregateFunctionTemplate<SingleValueDataGeneric, isMin...>(argument_types);
+}
+
+/// Helper to allocate enough memory to store any derived class
+struct SingleValueDataBaseMemoryBlock
+{
+    std::aligned_union_t<
```

Comment:
`aligned_union` is deprecated in C++23.

Reply:
And what’s the better alternative?

Reply:
Here is the document of C++ for the deprication: https://www.open-std.org/jtc1/sc22/wg21/docs/papers/2021/p1413r3.pdf 

It suggests the replacement as following:
```
// To replace std::aligned_union...
template <typename... Ts>
class MyContainer {
// [...]
private:
- std::aligned_union_t<0, Ts...> t_buff;
+ alignas(Ts...) std::byte t_buff[std::max({sizeof(Ts)...})];
// [...]
};
```

Reply:
Sure, we could do:
```c++
alignas(8) std::byte memory[SingleValueDataBase::MAX_STORAGE_SIZE];
```

And look for something similar, with tons of asserts, for Field::storage. Feel free to send a PR